xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -113,3 +118,108 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, 0
     )
+
+
+def _is_signed(transaction: Transaction) -> bool:
+    """
+    Checks if a transaction has been signed.
+
+    Args:
+        transaction: the transaction to check.
+
+    Returns:
+        Whether the transaction has been signed
+    """
+    return (
+        transaction.signing_pub_key is not None and len(transaction.signing_pub_key) > 0
+    ) or (transaction.txn_signature is not None and len(transaction.txn_signature) > 0)
+
+
+def _decode_tx_blob(tx_blob: str) -> Transaction:
+    """
+    Decodes a transaction blob.
+
+    Args:
+        tx_blob: the tx blob to decode.
+
+    Returns:
+        The formatted transaction.
+    """
+    return Transaction.from_xrpl(decode(tx_blob))
+
+
+async def _get_signed_tx(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction (or transaction blob) to be submitted.
+        client: the network client with which to submit the transaction.
+        wallet: the wallet with which to sign the transaction (optional, only needed
+        if the transaction is not signed).
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if isinstance(transaction, str):
+        transaction = _decode_tx_blob(transaction)
+
+    if _is_signed(transaction):
+        return transaction
+
+    if not wallet:
+        raise XRPLException(
+            "Wallet must be provided when submitting an unsigned transaction"
+        )
+
+    if check_fee:
+        await _check_fee(transaction, client)
```

Comment:
Isn't `check_fee` already run in `autofill`?

Reply:
`autofill` only auto-fills in the `fee` if the `fee` field in `transaction` is not yet defined, while the check in `check_fee` only happens when the input transaction is already pre-filled.

See the following comment in `autofill_and_sign`:
`# We do the transaction fee check here as we have the Client available.
    # The fee check will be done if transaction.fee exists. Otherwise the fee
    # will be auto-filled in autofill()`

Reply:
Should `check_fee` happen after `autofill`? (We are in control of the autofill function, but it seems to me like the validation should check the finalized version of the transaction instead of an in-between state, especially since autofill can change the fee)